File: FRONTEND/backgammon-frontend/src/app/shared/utils/constants/board.constants.ts
import { PlayerColor } from '../../../features/game-session/models/enums/player-color.enum';

export const BOARD_POINTS = 24;

export const BAR = 0;
export const OFF = -1;

export const HOME_BOARD = {
  [PlayerColor.White]: { start: 19, end: 24 },
  [PlayerColor.Black]: { start: 1, end: 6 },
};

export const WHITE_BEAR_OFF_TARGET = 25;
export const BLACK_BEAR_OFF_TARGET = 0;

export const MOVE_DIRECTION = {
  [PlayerColor.White]: 1,
  [PlayerColor.Black]: -1,
};

File: BACKEND/Domain/GameLogic/Constants/BoardConstants.cs
using Common.Enums.BoardState;

namespace Domain.GameLogic.Constants
{
    public static class BoardConstants
    {
        // Bar / Off
        public const int BarPosition = 0;
        public const int OffBoardPosition = -1;

        public const int BoardPoints = 24;

        // Home board ranges
        public static readonly (int Start, int End) WhiteHome = (19, 24);
        public static readonly (int Start, int End) BlackHome = (1, 6);

        // Move direction
        public const int WhiteDirection = 1; // 1 → 24
        public const int BlackDirection = -1; // 24 → 1

        // Bear off positions
        public const int WhiteBearOff = 25;
        public const int BlackBearOff = 0;

        public static bool IsHomeBoard(
            int point,
            PlayerColor player)
        {
            return player == PlayerColor.White
                ? point >= WhiteHome.Start && point <= WhiteHome.End
                : point >= BlackHome.Start && point <= BlackHome.End;
        }

        public static bool IsEntryPoint(
            int point,
            PlayerColor player)
        {
            return player == PlayerColor.White
                ? point >= BlackHome.Start && point <= BlackHome.End
                : point >= WhiteHome.Start && point <= WhiteHome.End;
        }

        public static int GetDirection(PlayerColor player)
            => player == PlayerColor.White ? WhiteDirection : BlackDirection;

        public static int GetBearOffTarget(PlayerColor player)
            => player == PlayerColor.White ? WhiteBearOff : BlackBearOff;
    }
}
